web/frontend/src/components/slam.vue

```diff
@@ -30,42 +30,60 @@ const loaded2d = $computed(() => (pointcloud !== undefined && pose !== undefined
 let slam2dIntervalId = -1;
 let slam3dIntervalId = -1;
 
+const concatArrayU8 = (arrays: Uint8Array[]) => {
+  const totalLength = arrays.reduce((acc, value) => acc + value.length, 0);
+  const result = new Uint8Array(totalLength);
+  let length = 0;
+  for (const array of arrays) {
+    result.set(array, length);
+    length += array.length;
+  }
+  return result;
+};
+
 const fetchSLAMMap = (name: string) => {
   return nextTick(() => {
-    const req = new slamApi.GetMapRequest();
+    const req = new slamApi.GetPointCloudMapStreamRequest();
     req.setName(name);
-    req.setMimeType('pointcloud/pcd');
-
     rcLogConditionally(req);
-    props.client.slamService.getMap(req, new grpc.Metadata(), (error, response) => {
-      if (error) {
-        return displayError(error);
-      }
-
-      /*
-       * TODO: This is a hack workaround the fac that at time of writing response!.getPointCloud()!.getPointCloud_asU8()
-       * appears to return the binary data of the entire response, not response.pointcloud.pointcloud.
-       * Tracked in ticket: https://viam.atlassian.net/browse/RSDK-2108
-       */
+    const chunks: Uint8Array[] = [];
 
-      const base64DecodedPointCloudString = atob(response!.toObject().pointCloud!.pointCloud! as string);
-      pointcloud = Uint8Array.from(base64DecodedPointCloudString, (char: string): number => char.codePointAt(0)!);
-      // END NOTE
+    const getPointCloudMapStream = props.client.slamService.getPointCloudMapStream(req);
+    getPointCloudMapStream.on('data', (res) => {
+      const chunk = res.getPointCloudPcdChunk_asU8();
+      chunks.push(chunk);
+    });
+    getPointCloudMapStream.on('status', (status) => {
+      if (status && status.code !== 0) {
```

Comment:
[q] is `status == false` (or the equivalent negative representation) ever returned?

Reply:
Also is `status.code = 200` considered a success?

Reply:
status.code === 0 is a success:
https://grpc.github.io/grpc/core/md_doc_statuscodes.html

It appears as if here https://github.com/viamrobotics/rdk/pull/1968/files/4401d890e8a7bdf135cd59f6dc0a231859ce3411#diff-8d6b59a2a5afc9dde0d409eca6b53c1b066221f1d729deddda801c0ca196bfffR67 `end` can be undefined, but here it can't so I've changed it.